icloudpd/base.py

```diff
@@ -516,7 +516,7 @@ def download_photo(counter, photo):
 
                 lp_file_exists = os.path.isfile(lp_download_path)
 
-                if only_print_filenames:
+                if only_print_filenames and not lp_file_exists:
```

Comment:
We need a test for this case.

Quickly scanning through the logic tree, I suspect that executing icloudpd with `--only-print-files` over duplicate live photo assets with different sizes where one already had been downloaded will cause unexpected side effects.

Reply:
I agree, but I did not come up with a nice test scenario... Ideas? I'm still not well up with these cassettes and stuff.

Actually I don't really like the structure in that section at all. Why that extra handling for Live fotos. Should work more generic and be refactored somewhen...

Reply:
Tests added. This was far more work than the fix itself... ;-) But a good invest in (future) quality.